usaspending_api/etl/management/commands/swap_in_new_table.py

```diff
@@ -0,0 +1,292 @@
+            self.swap_constraints_sql(cursor)
+            self.swap_table_sql(cursor)
+            if not options["keep_old_data"]:
+                self.drop_old_table_sql(cursor)
+            self.extra_sql(cursor)
+
+    def swap_index_sql(self, cursor):
+        logging.info("Renaming indexes of the new and old tables.")
+        temp_indexes = self.query_result_lookup["temp_table_indexes"]
+        curr_indexes = self.query_result_lookup["curr_table_indexes"]
+        rename_sql = []
+        sql_template = "ALTER INDEX {old_index_name} RENAME TO {new_index_name};"
+        for val in curr_indexes:
+            old_name = val["indexname"]
+            new_name = f"{old_name}_old"
+            rename_sql.append(sql_template.format(old_index_name=old_name, new_index_name=new_name))
+        for val in temp_indexes:
+            old_name = val["indexname"]
+            new_name = re.match("^(.*)_temp$", old_name, flags=re.I)[1]
+            rename_sql.append(sql_template.format(old_index_name=old_name, new_index_name=new_name))
+
+        if rename_sql:
+            cursor.execute("\n".join(rename_sql))
+
+    def swap_constraints_sql(self, cursor):
+        logging.info("Renaming constraints of the new and old tables.")
+        temp_constraints = self.query_result_lookup["temp_table_constraints"]
+        curr_constraints = self.query_result_lookup["curr_table_constraints"]
+        rename_sql = []
+        sql_template = "ALTER TABLE {table_name} RENAME CONSTRAINT {old_constraint_name} TO {new_constraint_name};"
+        for val in curr_constraints:
+            old_name = val["constraint_name"]
+            new_name = f"{old_name}_old"
+            rename_sql.append(
+                sql_template.format(
+                    table_name=self.db_table_name, old_constraint_name=old_name, new_constraint_name=new_name
+                )
+            )
+        for val in temp_constraints:
+            old_name = val["constraint_name"]
+            new_name = re.match("^(.*)_temp$", old_name, flags=re.I)[1]
+            rename_sql.append(
+                sql_template.format(
+                    table_name=self.temp_db_table_name, old_constraint_name=old_name, new_constraint_name=new_name
+                )
+            )
+
+        if rename_sql:
+            cursor.execute("\n".join(rename_sql))
+
+    @transaction.atomic
+    def swap_table_sql(self, cursor):
+        logging.info("Renaming the new and old tables")
+        sql_template = "ALTER TABLE {old_table_name} RENAME TO {new_table_name};"
+        rename_sql = [
+            sql_template.format(old_table_name=self.db_table_name, new_table_name=f"{self.db_table_name}_old"),
+            sql_template.format(old_table_name=self.temp_db_table_name, new_table_name=f"{self.db_table_name}"),
+        ]
+        cursor.execute("\n".join(rename_sql))
+
+    def drop_old_table_sql(self, cursor):
+        # Instead of using CASCADE, all old constraints and indexes are dropped manually
```

Comment:
Maybe expand on this comment. Is this because cascade can cause dropping constraints across multiple tables to all happen in 1 transaction, and can lead to deadlocks?